mkfs.c

```diff
@@ -274,6 +274,12 @@ int main(int argc, char **argv)
         stat_buf.st_size = blk_size;
     }
 
+    /* Initialize the file system image with zeros */
```

Comment:
You should clear a root index block instead. That is, `calloc(SIMPLEFS_BLOCK_SIZE)` and `write(fd, block, SIMPLEFS_BLOCK_SIZE)`. In addition, do verify the result of `write` system call.

Reply:
Hi @jserv, I think this issue is related to #20. The problem arises from incorrect reading of `simplefs_file_ei_block`. If the values inside blockdata are random or not the correct format, then the resulting `simplefs_file_ei_block` will be incorrect, causing the initial `nr_files` and `simplefs_extent` to have meaningless values. However, this leads to an `sb_bread` error during `simplefs_lookup` and other directory related problems.

To address this issue, the solution is to initialize the file system image with zeros instead of just clearing a root index block. This ensures that the values within blockdata are consistent and prevent the occurrence of random values. 
Alternatively, should I approach this issue differently to find a solution?

Reply:
> To address this issue, the solution is to initialize the file system image with zeros instead of just clearing a root index block. This ensures that the values within blockdata are consistent and prevent the occurrence of random values. Alternatively, should I approach this issue differently to find a solution?

I would expect the minimal operations against the raw file system image. That is, to clear a root index block instead of zeroing the whole file system image. You should investigate the necessary changes required to make the file system consistent.